apps/design/frontend/src/image_input.tsx

```diff
@@ -1,8 +1,61 @@
 import { Buffer } from 'buffer';
 import DomPurify from 'dompurify';
 import { FileInputButton } from '@votingworks/ui';
+import { assert, assertDefined } from '@votingworks/basics';
 
-const MAX_SVG_UPLOAD_BYTES = 5 * 1_000 * 1_000; // 5 MB
+const MAX_IMAGE_UPLOAD_BYTES = 5 * 1_000 * 1_000; // 5 MB
+
+const ALLOWED_IMAGE_TYPES = ['image/svg+xml', 'image/png', 'image/jpeg'];
+
+async function loadSvgImage(file: File): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      /* istanbul ignore next */
+      const contents = e.target?.result;
+      if (typeof contents === 'string') {
+        resolve(contents);
+      }
+      reject(new Error('Could not read file contents'));
+    };
+    reader.readAsText(file);
+  });
+}
+
+async function getBitmapImageDimensions(
+  imageDataUrl: string
+): Promise<{ width: number; height: number }> {
+  return new Promise((resolve) => {
+    const img = new Image();
+    img.onload = () => {
+      resolve({ width: img.naturalWidth, height: img.naturalHeight });
+    };
+    img.src = imageDataUrl;
+  });
```

Comment:
This version will ignore errors and never resolve. Use the [`Image::decode`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/decode) method that I just learned about which will reject on errors?

Reply:
thanks! `Image::decode` is great